_plugins/locale_url_filter.rb

```diff
@@ -2,14 +2,23 @@ module Jekyll
   module LocaleUrlFilter
     LOCALE_STRING_CONSTANT = '__LOCALE_BASE_URL__'.freeze
 
-    def locale_url(path = '')
+    def locale_url(path = '', locale = nil)
       site_base_url = @context.registers[:site].baseurl
-      locale_prefix = @context.registers[:page]['base_path']
-      site_base_url.to_s + locale_prefix.to_s + path
+      locale ||= @context.registers[:page]['lang']
+      permalink = @context.registers[:site].collections[locale].metadata['permalink']
+      site_base_url.to_s + Jekyll::URL.new(
+        template: permalink,
+        placeholders: { collection: locale, path: path },
+      ).to_s
+    end
+
+    def delocalize_url(path, locale = nil)
+      locale ||= @context.registers[:page]['lang']
+      path.gsub(%r{^/#{locale}/}, '/')
     end
 
     def replace_locale_base_url(input)
-      input.gsub LOCALE_STRING_CONSTANT, locale_url
+      input.gsub(LOCALE_STRING_CONSTANT, locale_url.gsub(%r{/$}, ''))
```

Comment:
Also I realized this after I pushed, I think we can do this without regexes

```suggestion
      input.gsub(LOCALE_STRING_CONSTANT, locale_url.chomp('/'))
```

Reply:
> Also I realized this after I pushed, I think we can do this without regexes

Good call 👍  Updated in 9764290.